src/dash-playlist-loader.js

```diff
@@ -204,7 +206,44 @@ export const requestSidx_ = (sidxRange, playlist, xhr, options, finishProcessing
     headers: segmentXhrHeaders(sidxInfo)
   });
 
-  return xhr(sidxRequestOptions, finishProcessingFn);
+  return containerRequest(sidxInfo.uri, xhr, (err, request, container, bytes) => {
+    if (err) {
+      return finishProcessingFn(err, request);
+    }
+
+    if (!container || container !== 'mp4') {
+      this.error = {
```

Comment:
Because this is a standalone function, even though it can be `call`ed with a context, I think it would be preferable to pass in an error callback handler or use some other system than to use `this`.